src/food/food.controller.ts

```diff
@@ -24,102 +24,111 @@ export class FoodController {
-  })
-  @ApiCreatedResponse({ description: '음식 list', type: Food })
+  @ApiOperation({ summary: '테스트 음식 리스트를 가져오는 API' })
   async findTestFoodList(): Promise<Food[]> {
-    return this.foodService.findTestFoods();
+    try {
+      const foods: Promise<Food[]> = this.foodService.findTestFoods();
+      return foods;
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
   }
 
   @Post()
-  @ApiOperation({
-    summary: '음식 정보를 저장하는 API',
-    description: '새로운 음식 정보를 저장합니다.',
-  })
-  @ApiBody({ type: CreateFoodDto })
-  @ApiCreatedResponse({ description: '등록 된 음식 정보', type: CreateFoodDto })
+  @ApiOperation({ summary: '음식 정보를 저장하는 API' })
   async createFoodInfo(
     @Body() createFoodDto: CreateFoodDto,
   ): Promise<CreateFoodDto> {
-    return this.foodService.createFoodInfo(createFoodDto);
+    const { name, subName, level, category } = createFoodDto;
+    try {
+      const foodLevel = await this.foodService.findFoodLevelByLevelId(level);
+      const foodId = await this.foodService.inputFood(name, subName, foodLevel);
+      const food = this.foodService.setFoodCategory(
+        name,
+        subName,
+        level,
+        category,
+        foodId,
+      );
+
+      return food;
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
   }
 
   @Get('/tests/result')
-  @ApiOperation({
-    summary: '사용자 레벨별 음식 추천 API',
-    description: '사용자 레벨별 음식을 추천합니다.',
-  })
-  @ApiQuery({
-    name: 'userLevel',
-    type: FindUserLevelFoodDto,
-    description: '사용자 레벨',
-  })
-  @ApiCreatedResponse({ description: '레벨별 음식 정보', type: Food })
+  @ApiOperation({ summary: '사용자 레벨별 음식 추천 API' })
   async findUserLevelFoods(
     @Query() param: FindUserLevelFoodDto,
   ): Promise<Food[]> {
-    return this.foodService.findUserLevelFoods(param);
+    const { userLevel } = param;
+    try {
+      const { id: foodLevel } = await this.foodService.findUserLevelFoods(
+        userLevel,
+      );
+      const treeFood = this.foodService.findTreeFood(foodLevel);
+      return treeFood;
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
   }
 
   @Get('/random')
-  @ApiOperation({
-    summary: '음식 랜덤 추천 API',
-    description: '음식을 랜덤 추천합니다.',
-  })
-  @ApiQuery({ name: 'userId', type: String, description: '사용자 ID' })
-  @ApiResponse({ description: '레벨별 음식 정보', type: RandomFoodDto })
+  @ApiOperation({ summary: '음식 랜덤 추천 API' })
   async findRandomFood(
-    @Query() param: { userId: string },
+    @Query() param: FindUserLevelByUserIdDto,
   ): Promise<RandomFoodDto> {
-    return this.foodService.findRandomFoods(param.userId);
+    const { userId } = param;
+    try {
+      const userlevel = await this.foodService.findUserLevel(userId);
+
+      if (userlevel.id === '5') userlevel.id = '4';
+
+      const food = await this.foodService.findFoodByUserLevel(userlevel);
+
+      return food;
```

Comment:
다른곳은 await 없이 return했는데 여기만 await을 한 이유가 있을까요??

```suggestion
     return this.foodService.findFoodByUserLevel(userlevel);
```